### ISSUE TITLE
FW Additional Entries

### ISSUE THREAD

**Issue by agconstantin** [RedHatGov/ssg-el7-kickstart#48]
Hello,
I am trying to add new firewall rules into the final ISO image, but seems they are not taking into consideration. Even the SSH port is not opened and needs to be readeed, once the new server is deployed.
So far I have tried to add the firewall-cmd rules to the following files:
ssg-rhel.cfg
ssg-supplemental.sh
menu.py

None is taking into consideration the new FW entries.
Any suggestion on this one?
Thank you.

**Comment by fcaviggia** [comment]
How are you adding the firewall rules?

Typically, for `firewall-cmd` it's a two part configuration.  In the example below - the first command adds the rule to the permanent configuration second command reloads the configuration:

```
# firewall-cmd --permanent --add-service=ssh
# firewall-cmd --reload
```

**Comment by agconstantin** [comment]
Hi Frank,
That is exactly how I do it, still they are not in effect after deploying from the generated ISO.
Any clues why?

Regards,
Alin

**Comment by fcaviggia** [comment]
I'll try and check into it tonight...

**Comment by agconstantin** [comment]
Any update on this one?

**Comment by fcaviggia** [comment]
I think there were some profiles where I used iptables/ip6tables/ebtables and some where I used firewalld - that might be part of the problem. I'm going to have to think about how I want to deploy customization for profiles (e.g. additional RPMs, firewall changes, etc.)  I'll try to work on this here at some point.

**Comment by agconstantin** [comment]
If you tell me where you make the call for the firewall rules, I can try to fix on that using the firewalld.

**Comment by jmaughmer** [comment]
The command you would use for changing the firewall in the kickstarter is actually firewall-offline-cmd. I have made some customizations to some of the scripts and I add my firewall configuration just before %end in ssg-rhel.cfg

## Firewall configuration
# SSH
firewall-offline-cmd --zone=drop --add-service=ssh

**Comment by agconstantin** [comment]
@jmaughmer have you tested that? because I have tried that in multiple places/references including ssg-rhel.cfg, without been applied to the generated ISO image.

**Comment by jmaughmer** [comment]
Yes, It has worked with RHEL 7.3 and 7.4 server and workstation. This is what I have in ssg-rhel.cfg at the end right before the %end.

## Firewall configuration
# SSH
firewall-offline-cmd --zone=drop --add-service=ssh
# Jabber/OSAD
firewall-offline-cmd --zone=drop --add-port=5222/tcp
# McAfee ePO Agent
firewall-offline-cmd --zone=drop --add-port=8081/tcp
firewall-offline-cmd --zone=drop --add-port=8082/tcp
#ICMP needed for Nessus
firewall-offline-cmd --direct --add-rule ipv4 filter INPUT 0 -p icmp -s 0.0.0.0/0 -d 0.0.0.0/0 -j ACCEPT

Both Server and Workstation has this in the firewall rules after install:

Chain IN_drop_allow (1 references)
target     prot opt source               destination         
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:22 ctstate NEW
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:5222 ctstate NEW
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:8081 ctstate NEW
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:8082 ctstate NEW
